In [21]:
import pandas as pd
import numpy as np


# Section 1: Exploratory Data Analysis

### All questions in this section pertain to two of the datasets listed above in Exam Materials 

### (person_level_data.csv and household_level_data.csv), with the data_dictionary.txt file providing the metadata needed to understand the contents and formats of each .csv. 


### Please note that the data is still in its raw, original form and will require some cleaning and/or standardization.


Please write your answers in this document, and include your code as part1_code.txt.
Questions


Please calculate the following summary statistics using the data sets listed above:


1.	Calculate the average weekly hours worked among men and women.
2.	List the three households (i.e. household_ids) with the highest per person incomes. 
3.	List the three individuals (i.e. person_ids) with the highest personal income, and the dollar amounts of their personal incomes.
4.	Calculate the 80th percentile of age among people in each of the five marital categories.


In [22]:
person = pd.read_csv('person_level_data.csv')
household = pd.read_csv('household_level_data.csv')

person.head()

person_id  age citizen_status nativity_status  weekly_hours_worked  \
0         10   62  citizen_birth     native_born                   55   
1         11   65  citizen_birth     native_born                   40   
2         15   70  citizen_birth     native_born                    0   
3         16   70  citizen_birth     native_born                   20   
4         43   75  citizen_birth     native_born                    0   

  marital_status school_status     sex     when_last_worked worked_last_week  \
0        married   not_student  FEMALE         in_last_year           worked   
1        married   not_student       M         in_last_year           worked   
2        married   not_student    MALE  over_five_years_ago     did_not_work   
3        married   not_student       F         in_last_year     did_not_work   
4        married   not_student  FEMALE  over_five_years_ago     not_reported   

   ... race_native_american  race_asian  race_black  race_native_hawaiian  \
0  ...                    0           0           0                     0   
1  ...                    0           0           0                     0   
2  ...                    0           0           0                     0   
3  ...                    0           0           0                     0   
4  ...                    0           0           0                     0   

   race_pacific_islander  race_white  race_other  uninsured  household_id  \
0                      0           1           0          0             5   
1                      0           1           0          0             5   
2                      0           1           0          0             9   
3                      0           1           0          0             9   
4                      0           1           0          0            22   

   hh_income_pct  
0       0.389078  
1       0.610922  
2       0.735381  
3       0.264619  
4       0.250230  

[5 rows x 21 columns]

In [23]:
# person.weekly_hours_worked.value_counts()

In [24]:
person.sex.value_counts()

FEMALE    7223
MALE      6735
F         2056
M         1930
female    1017
male       936
Name: sex, dtype: int64

In [25]:
# Combining the different values for male and female
person.sex = person.sex.replace({'FEMALE': 'F'})
person.sex = person.sex.replace({'female': 'F'})
person.sex = person.sex.replace({'MALE': 'M'})
person.sex = person.sex.replace({'male': 'M'})
person.sex.value_counts()


F    10296
M     9601
Name: sex, dtype: int64

In [26]:
person.head()

person_id  age citizen_status nativity_status  weekly_hours_worked  \
0         10   62  citizen_birth     native_born                   55   
1         11   65  citizen_birth     native_born                   40   
2         15   70  citizen_birth     native_born                    0   
3         16   70  citizen_birth     native_born                   20   
4         43   75  citizen_birth     native_born                    0   

  marital_status school_status sex     when_last_worked worked_last_week  ...  \
0        married   not_student   F         in_last_year           worked  ...   
1        married   not_student   M         in_last_year           worked  ...   
2        married   not_student   M  over_five_years_ago     did_not_work  ...   
3        married   not_student   F         in_last_year     did_not_work  ...   
4        married   not_student   F  over_five_years_ago     not_reported  ...   

  race_native_american  race_asian  race_black  race_native_hawaiian  \
0                    0           0           0                     0   
1                    0           0           0                     0   
2                    0           0           0                     0   
3                    0           0           0                     0   
4                    0           0           0                     0   

   race_pacific_islander  race_white  race_other  uninsured  household_id  \
0                      0           1           0          0             5   
1                      0           1           0          0             5   
2                      0           1           0          0             9   
3                      0           1           0          0             9   
4                      0           1           0          0            22   

   hh_income_pct  
0       0.389078  
1       0.610922  
2       0.735381  
3       0.264619  
4       0.250230  

[5 rows x 21 columns]

## Calculate the average weekly hours worked among men and women.

In [27]:
person.groupby('sex').weekly_hours_worked.mean()

sex
F    21.725039
M    28.669930
Name: weekly_hours_worked, dtype: float64

### On average men worked 28.67 hours

### On average women worked 21.73 hours

## List the three households (i.e. household_ids) with the highest per person incomes.

In [28]:
household.head()


household_id      variable    value
0             5     hh_income  90334.0
1             5       hh_size      2.0
2             5  num_children      0.0
3             9     hh_income  66216.0
4             9       hh_size      2.0

In [29]:
# household = household.pivot(index='variable', columns='household_id', values='value')


# household

In [30]:
household = household.T

household.columns

household

0        1             2          3        4      \
household_id          5        5             5          9        9   
variable      hh_income  hh_size  num_children  hh_income  hh_size   
value             90334        2             0      66216        2   

                     5          6        7             8          9      ...  \
household_id             9         22       22            22         31  ...   
variable      num_children  hh_income  hh_size  num_children  hh_income  ...   
value                    0      84614        3             0      60285  ...   

                     32948      32949    32950         32951      32952  \
household_id         55139      55140    55140         55140      55141   
variable      num_children  hh_income  hh_size  num_children  hh_income   
value                    0      51551        2             0      84673   

                32953         32954      32955    32956         32957  
household_id    55141         55141      55148    55148         55148  
variable      hh_size  num_children  hh_income  hh_size  num_children  
value               2             0      41676        1             0  

[3 rows x 32958 columns]

In [31]:
household ['per_person'] = household.hh_income/household.hh_size

household.sort_values(by='per_person', ascending=False)

AttributeError: 'DataFrame' object has no attribute 'hh_income'

###  The three households with the highest per person incomes found by dividing the total income by family size are,

### household id, 38870
### household id, 24934
### household id, 10110

## List the three individuals (i.e. person_ids) with the highest personal income, and the dollar amounts of their personal incomes.

Personal id = 23355, Income = $1060273


Personal id = 89429, Income = $840245


Personal id = 82159, Income = $815297

In [ ]:
pd.set_option('display.max_columns', 500)
person.head()

hh_income_pct:
  description:
    - Percentage of the household's total income that comes from that person; 
      calculated as (person's total income) / (household income).

the total income earned per person; this can
be calculated, however, by multiplying the person’s percent of household
income (hh_income_pct) with the household income (hh_income) found in the
household-level data set.

In [ ]:
new = pd.merge(person, household, how='inner', on='household_id')

new.head()

In [ ]:
new['personal_income'] = new.hh_income_pct*new.hh_income

new = new.sort_values(by='personal_income', ascending=False)
new.head(3)

Personal id = 23355, Income = $1060273


Personal id = 89429, Income = $840245


Personal id = 82159, Income = $815297

## Calculate the 80th percentile of age among people in each of the five marital categories.


divorced         67.0


married          67.0


never_married    45.0


separated        61.0


widowed          86.0

In [ ]:
new.marital_status.value_counts()


In [32]:
new.groupby('marital_status').age.quantile(0.8)


NameError: name 'new' is not defined